src/html.rs

```diff
@@ -137,24 +140,28 @@ impl HtmlMsgParser {
                     let mimetype = mail.ctype.mimetype.parse::<Mime>()?;
                     if mimetype == mime::TEXT_HTML {
                         if let Ok(decoded_data) = mail.get_body() {
-                            self.html = decoded_data;
+                            if self.html.is_empty() {
+                                self.html = decoded_data;
+                            }
                             return Ok(());
                         }
                     } else if mimetype == mime::TEXT_PLAIN {
                         if let Ok(decoded_data) = mail.get_body() {
-                            self.plain = Some(PlainText {
-                                text: decoded_data,
-                                flowed: if let Some(format) = mail.ctype.params.get("format") {
-                                    format.as_str().to_ascii_lowercase() == "flowed"
-                                } else {
-                                    false
-                                },
-                                delsp: if let Some(delsp) = mail.ctype.params.get("delsp") {
-                                    delsp.as_str().to_ascii_lowercase() == "yes"
-                                } else {
-                                    false
-                                },
-                            });
+                            if self.plain.is_none() {
```

Comment:
Same as above, this `if` could be swapped with previous one.

Reply:
i changed that.